From 26af2265232b7c7303f18eca9edb005bd5efb357 Mon Sep 17 00:00:00 2001
From: Tom Kirkpatrick <tkp@kirkdesigns.co.uk>
Date: Fri, 26 Jul 2019 14:49:00 +0100
Subject: [PATCH] lnd: expose user agent version as config option

Expose the neutrino `UserAgentVersion` conffig option. This can be set
by starting lnd with the `--neutrino.useragentversion=` flag.
---
 chainregistry.go | 1 +
 config.go        | 1 +
 docs/INSTALL.md  | 1 +
 3 files changed, 3 insertions(+)

diff --git a/chainregistry.go b/chainregistry.go
index e637c560c1..86a5dbc836 100644
--- a/chainregistry.go
+++ b/chainregistry.go
@@ -753,6 +753,7 @@ func initNeutrinoBackend(chainDir string) (*neutrino.ChainService, func(), error
 	neutrino.MaxPeers = 8
 	neutrino.BanDuration = time.Hour * 48
 	neutrino.UserAgentName = cfg.NeutrinoMode.UserAgentName
+	neutrino.UserAgentVersion = cfg.NeutrinoMode.UserAgentVersion
 
 	neutrinoCS, err := neutrino.NewChainService(config)
 	if err != nil {
diff --git a/config.go b/config.go
index 6691c0d8b0..94f515879d 100644
--- a/config.go
+++ b/config.go
@@ -182,6 +182,7 @@ type neutrinoConfig struct {
 	FeeURL             string        `long:"feeurl" description:"Optional URL for fee estimation. If a URL is not specified, static fees will be used for estimation."`
 	AssertFilterHeader string        `long:"assertfilterheader" description:"Optional filter header in height:hash format to assert the state of neutrino's filter header chain on startup. If the assertion does not hold, then the filter header chain will be re-synced from the genesis block."`
 	UserAgentName      string        `long:"useragentname" description:"Used to help identify ourselves to other bitcoin peers"`
+	UserAgentVersion   string        `long:"useragentversion" description:"Used to help identify ourselves to other bitcoin peers"`
 }
 
 type btcdConfig struct {
diff --git a/docs/INSTALL.md b/docs/INSTALL.md
index 661223d468..6b485e4c6b 100644
--- a/docs/INSTALL.md
+++ b/docs/INSTALL.md
@@ -194,6 +194,7 @@ neutrino:
       --neutrino.banduration=                                 How long to ban misbehaving peers.  Valid time units are {s, m, h}.  Minimum 1 second
       --neutrino.banthreshold=                                Maximum allowed ban score before disconnecting and banning misbehaving peers.
       --neutrino.useragentname=                               Used to help identify ourselves to other bitcoin peers.
+      --neutrino.useragentversion=                            Used to help identify ourselves to other bitcoin peers.
 ```
 
 ## Bitcoind Options
